### ISSUE TITLE
Can anyone confirm this working?

### ISSUE THREAD

**Issue by BCITMike** [rodeodomino/algo-pihole#28]
I had a working install of Algo server on a KVM VPS but I wanted to share the pihole white and blacklists that I have in use at home and elsewhere.

So I reinstalled Ubuntu 16.04 minimal and then upgraded to Ubuntu 18.04 and did the local install of Algo. Added my clients and installation finished successfully with no errors.

However, I have had no success getting any clients to pass traffic and would like to know if anyone else can confirm this working for them?

When I try to connect the split tunnel wireguard into Android 9, I get `Error bringing up tunnel: Bad address`. 
I can connect using the full wireguard, but unable to pass traffic. I can load the pihole admin page from my phone using 172.26.237.42 but it shows that FTL is Offline. When I check the status of FTL, it has a conflict with starting on port 53.
```
# systemctl status pihole-FTL -l                                                                                                                                                                            
● pihole-FTL.service - LSB: pihole-FTL daemon
   Loaded: loaded (/etc/init.d/pihole-FTL; generated)
   Active: active (exited) since Tue 2019-09-03 01:08:55 BST; 5s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 1455 ExecStop=/etc/init.d/pihole-FTL stop (code=exited, status=0/SUCCESS)
  Process: 1470 ExecStart=/etc/init.d/pihole-FTL start (code=exited, status=0/SUCCESS)

Sep 03 01:08:55 sea.505090.net systemd[1]: Starting LSB: pihole-FTL daemon...
Sep 03 01:08:55 sea.505090.net pihole-FTL[1470]: Not running
Sep 03 01:08:55 sea.505090.net su[1516]: Successful su for pihole by root
Sep 03 01:08:55 sea.505090.net su[1516]: + ??? root:pihole
Sep 03 01:08:55 sea.505090.net su[1516]: pam_unix(su:session): session opened for user pihole by (uid=0)
Sep 03 01:08:55 sea.505090.net pihole-FTL[1470]: dnsmasq: failed to create listening socket for port 53: Address already in use
Sep 03 01:08:55 sea.505090.net su[1516]: pam_unix(su:session): session closed for user pihole
Sep 03 01:08:55 sea.505090.net systemd[1]: Started LSB: pihole-FTL daemon.
```
```
# netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      853/lighttpd        
tcp        0      0 127.0.2.1:53            0.0.0.0:*               LISTEN      1/init              
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      810/sshd            
tcp        0      0 172.26.xxx.yy:5353      0.0.0.0:*               LISTEN      705/dnscrypt-proxy  
udp        0      0 127.0.2.1:53            0.0.0.0:*                           1/init              
udp        0      0 0.0.0.0:68              0.0.0.0:*                           588/dhcpcd          
udp        0      0 172.26.xxx.yy:5353      0.0.0.0:*                           705/dnscrypt-proxy  
udp        0      0 0.0.0.0:4500            0.0.0.0:*                           789/charon          
udp        0      0 0.0.0.0:500             0.0.0.0:*                           789/charon          
udp        0      0 0.0.0.0:51820           0.0.0.0:*                           -                   
```

I think we need to see 172.26.xxx.yy:53 with FTL (something or another) for this to be working.

On Windows 10, I successfully imported the .ps1, but when I try and connect:
`Can't connect to AlgoVPN algo-pihole IKEv2`
`Policy match error`

So following the instructions without any obvious errors or deviation from the norm (public IPv4 VPS running Ubuntu 18.04) leads to non functioning Android and Windows (what I can test).

It would help if someone can confirm this working for them, in which case something is wrong on my end. But if others are having problems, that lets me know to move back to what 'just worked' before.

Thanks

**Comment by BCITMike** [comment]
Testing on another server, I was able to install it and test that it was working on Android. None of my public VPS' or home machines have IPv6 connectivity, so I always disable it in grub to avoid issues with IPv6.

When I disabled IPv6 in grub and rebooted,, my wireguard client could still connect but not pass traffic.
I then reenabled IPv6 in grub, rebooted, and now I can see it passing traffic again.

I'm about to do another install with the config.cfg edited to remove IPv6, will see if that flies.

**Comment by BCITMike** [comment]
So I reinstalled the server and retried algo-pihole and this time I was successful.

Things that were different:
1. It looks like systemd's resolvd was binding to port 53. Fix was to: `echo DNSStubListener=no >> /etc/systemd/resolved.conf` when I was on Ubuntu 16.04 and then upgraded to 18.04.
2. I left IPv6 enabled in Ubuntu and in the algo script. After installation and Android verified connected in Wireguard, I disabled IPv6 in Ubuntu and rebooted. When I reconnected the client, it did seem like things weren't working initially, but maybe after some timeout (10s, 30s?), the browser started working and seems to still be working. *shrug*
3. The crontab mentioned in the readme didn't work for me. When I changed it to: `*/5 * * * * /usr/local/bin/pihole-cloudsync/pihole-cloudsync --push > /dev/null 2>&1`, then it worked.


On another VPS, I disabled IPv6, and in the algo script I commented out all the IPv6 stuff, and the script failed when I went to install it.  `fatal: [localhost]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'local_service_ipv6' is undefined"}`

Overall, I think the issue was systemd's resolvd binding to port 53, NOT IPv6 being disabled in Ubuntu.

Hope this helps someone else.
